components/Modals/SignUp.tsx

```diff
@@ -50,6 +68,7 @@ export default function SignUpModal({ open, onClose }) {
         //The following line saves the latest localStorage progress to the backend.
         saveProgress(progress)
         onClose()
+        onSignUpComplete && saveProgress(getNextLessonKey(progress))
```

Comment:
Got it working! @afterburn I'd like one more re-review just to confirm this is an ok method to go about it,
changes made here...
Once the modal closes i save the progress to the next lessonkey whereby this allows the user to then proceed to the next key. I decided to not make a switch whereby the normal `saveProgress()` would not run if onSignUpComplete is not true because I felt that this ensures a linear progress path, i.e. no skipping progress in case there is an error.
Let me know what you think

Reply:
What exactly is the problem? If a user completes chapter 1, and then registers, we should set his/her progress key to `CH2INT1`. I'm guessing thats what you are attempting to do here with `saveProgress(getNextLessonKey(progress))` right?

Reply:
it instead saves the local progress to `CH2INT1` but the backend gets saved to `CH1OUT2`

Reply:
Ok, I went ahead and tried `saveProgress(onSignupComplete ? nextLessonKey : progress)` after initializing nextLessonKey as `const nextLessonKey = getNextLessonKey(progress)`
And that seems to have worked